### PR TITLE
Upgrade to JupyterLab 3

### PR DESCRIPTION
**Pull request by mbektas**
- Upgrades to JupyterLab 3.1.10 version
- MVP solution for stand-alone and self-contained desktop application for macOS, Debian, Ubuntu, Red Hat, Fedora and Windows with GUI installers for each platform
- Bundles JupyterLab server using conda constructor and automatically installs the conda environment during app installation. Following conda packages are included in the environment:
```
  - python 3.8
  - numpy
  - scipy
  - pandas
  - jupyterlab 3.1.10
  - ipywidgets 7.6.4
  - matplotlib
```
- Some features are removed (remote JupyterLab server connection etc) for now with the initial goal being providing an MVP desktop app

macOS
jlab-mac-install

jlab-mac

Ubuntu
jlab-ubuntu-install

jlab-ubuntu

Fedora
jlab-fedora-install

jlab-fedora

Windows
jlab-windows-install

jlab-windows

### CODE REVIEW

**Comment by mbektasbbg** [comment]
Current state:
jlab desktop app

**Comment by ellisonbg** [comment]
This is awesome! Somewhere we have the account, certificates, and information needed to create the app for MacOS. Let me know if you want to dive into that and I can track down the needed information.

**Comment by mbektasbbg** [comment]
> This is awesome! Somewhere we have the account, certificates, and information needed to create the app for MacOS. Let me know if you want to dive into that and I can track down the needed information.

Thanks @ellisonbg ! Sure, please share that information with me. Not so urgent but I will need those.

**Comment by echarles** [comment]
Are the instructions on the README still valid to test this PR?

**Comment by mbektasbbg** [comment]
> Are the instructions on the README still valid to test this PR?

Yes @echarles, yarn commands work for me. I didn't update & test distribution build yet. Are you having any issues?

`yarn install` and `yarn build:all` should work fine

**Comment by echarles** [comment]
@mbektasbbg No issue, `yarn` and `yarn build:all` work fine on my env and the electron window is launched correctly. 

I am on mac, and confused with `To build for macOS a macOS computer is required.`: There is no `yarn dockerdist:macos`target.

Then there is  `If you don't want to user Docker but instead want to build locally, there are a few dependencies you're required to install.`: On https://github.com/electron-userland/electron-builder I read `To ensure your native dependencies are always matched electron version, simply add script "postinstall": "electron-builder install-app-deps" to your package.json.`. What should I do? (As you see, first-time electron developer here).

**Comment by mbektasbbg** [comment]
> I am on mac, and confused with `To build for macOS a macOS computer is required.`: There is no `yarn dockerdist:macos`target.

@echarles readme seems to be pretty outdated. Once I update distribution related code, I will make sure readme is up to date as well.

**Comment by echarles** [comment]
Thx a lot @mbektasbbg for working on this. Looking forward the instructions when you will publish them.

**Comment by fperez** [comment]
Thx so much @mbektasbbg for this work!! Just yesterday I was discussing this app with @lgfunderburk, in the context of her work at [Cybera/Callysto](https://www.callysto.ca). For some of her use cases, a standalone, self-contained, "native" app would be very valuable. I missed your recent work on it, so it's great to see it moving forward!

**Comment by mbektasbbg** [comment]
> Thx so much @mbektasbbg for this work!! Just yesterday I was discussing this app with @lgfunderburk,

Thanks @fperez! Yes, our goal is to provide a self-contained native app which would also be able to connect to remote Jupyter server backends.

**Comment by mbektas** [comment]
> Thx a lot @mbektasbbg for working on this. Looking forward the instructions when you will publish them.

@echarles you can now follow the updated instructions to create a Mac installer and test. Now we have a self-contained app with its own conda env. bundled in. More updates to follow.

**Review by blink1073 — APPROVED** [review]
LGTM!  Happy to merge and iterate, but let's leave this open for a day in case anyone else has feedback

**Comment by blink1073** [comment]
The yarn file has a conflict

**Review by jtpio — APPROVED** [review]
This is great, thanks @mbektas for the update!

**Comment by jtpio** [comment]
Given that this is clearly a big improvement compared to the current state on `master`, we can indeed go ahead with a merge and iterate :+1:

File: LICENSE [hunk 1]
```diff
@@ -27,3 +25,9 @@ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
 CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+Semver File License
+===================
+
+The semver.py file is from https://github.com/podhmo/python-semver
+which is licensed under the "MIT" license.  See the semver.py file for details.
```
Comment (krassowski) [inline LICENSE:33]: Where is this file?

Comment (mbektas) [reply LICENSE:33]: @krassowski I copied the LICENSE file as is from jupyterlab repo. I just checked, semver.py is shipped with bundled jupyterlab python package. but I am not sure if it is OK to remove from jupyterlab-desktop license.

**Comment by blink1073** [comment]
> but I am not sure if it is OK to remove from jupyterlab-desktop license

I'd say it is OK to remove, since that file was never explicitly included in this repo.

**Pull request merged**